src/actions/save/index.js

```diff
@@ -1,17 +1,46 @@
 /* @flow */
-
-type SaveActionHandlers = {|
+import { ValidationError } from "../../lib"
+type SaveActionConfig = {|
   // eslint-disable-next-line no-restricted-globals, promise/no-native
-  onCreateVaultToken: () => Promise<string>,
-  onApprove: ({| token: string |}) => void,
+  createVaultSetupToken: () => Promise<string>,
+  onApprove: ({| vaultSetupToken: string |}) => void,
 |};
 
-export type SaveAction = (SaveActionHandlers) => ({|
+export type SaveAction = (SaveActionConfig) => ({|
   type: "save",
-  ...SaveActionHandlers,
+  /* TODO: 
+      - We need to define how paymentSourceDetails is typed here
+      - Do we have another `onError` type from elsewhere we could reference here?
+  */
+  save: ({|onError: () => {}, paymentSourceDetails: any|}) => Promise<void>,
 |});
 
-export const createSaveAction: SaveAction = (handlers: SaveActionHandlers) => ({
-  type: "save",
-  ...handlers,
-});
+/**
+ * These are the input configurations required from the merchant.
+ */
+const REQUIRED_INPUTS = {
+  onApprove: 'function',
+  createVaultSetupToken: 'function'
+}
+
+const validateSaveConfig = (config: SaveActionConfig) => {
+  for (const [inputProp, inputType] of Object.entries(REQUIRED_INPUTS)) {
+    if (!config[inputProp] || typeof config[inputProp] !== inputType) {
+      throw new ValidationError(`Save action is missing the required '${inputProp}' callback`)  
+    }
+  }
+}
+
+/**
+ * Creating a Save action allows us to validate initial inputs from the merchant, and then return the resulting object. 
+ */
+export const createSaveAction: SaveAction = (config: SaveActionConfig) => {
+  const { createVaultSetupToken } = config;
```

Comment:
Would we want to validate prior to destructuring anything off of the config? Maybe it's a bit early to get that picky about the code?

Reply:
I need to push some changes, but yeah this is just left from when I did it here initially and destructured the props. I forgot to delete this line when I moved validation to a function, so yeah it'll be removed here shortly